### PR TITLE
feat(fgs/function): support version aliases management

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
An alias points to a specific function version. Create an alias and expose it to clients, for example, bind a trigger to the alias instead of the corresponding version.
In the feature, the version management will be supported.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support version aliases management.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_versions'==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_versions -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_versions
=== PAUSE TestAccFgsV2Function_versions
=== CONT  TestAccFgsV2Function_versions
--- PASS: TestAccFgsV2Function_versions (25.49s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       25.579s
```

### CODE REVIEW

File: docs/resources/fgs_function.md [hunk 1]
```diff
@@ -76,6 +76,31 @@ resource "huaweicloud_fgs_function" "by_swr_image" {
 }
 ```
 
+### Create an alias for latest version
```
Comment (deer-hang) [inline docs/resources/fgs_function.md]: ### Create function with an alias for latest version

Comment (Lance52259) [reply docs/resources/fgs_function.md]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_basic -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_basic
=== PAUSE TestAccFgsV2Function_basic
=== CONT  TestAccFgsV2Function_basic
--- PASS: TestAccFgsV2Function_basic (61.64s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       61.731s
```
```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_versions'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_versions -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_versions
=== PAUSE TestAccFgsV2Function_versions
=== CONT  TestAccFgsV2Function_versions
--- PASS: TestAccFgsV2Function_versions (36.21s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       36.309s
```

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go [hunk 2]
```diff
@@ -427,19 +494,91 @@ func flattenFgsCustomImage(imageConfig function.CustomImage) []map[string]interf
...
+			})
+		} else {
+			result[v.Version] = []interface{}{
+				map[string]interface{}{
+					"name":        v.Name,
+					"description": v.Description,
+				},
+			}
+		}
+	}
```
Comment (deer-hang) [inline huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go:535]:
```suggestion
	for _, v := range aliasList {
		result[v.Version] = append(result[v.Version], map[string]interface{}{
			"name":        v.Name,
			"description": v.Description,
		})
	}
```

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go [hunk 3]
```diff
@@ -362,9 +400,38 @@ func resourceFgsFunctionV2Create(d *schema.ResourceData, meta interface{}) error
...
+func createFunctionVersions(client *golangsdk.ServiceClient, functionUrn string, versionSet *schema.Set) error {
+	for _, v := range versionSet.List() {
+		version := v.(map[string]interface{})
+		versionNum := version["name"].(string) // The version name, also name as the version number.
+		// In the future, the function will support manage multiple versions, and will add the corresponding logic to
+		// create versions based on the related API (Create) in this place.
+		aliasCfg := version["aliases"].([]interface{})
+		if len(aliasCfg) < 1 {
+			continue
+		}
```
Comment (deer-hang) [inline huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go:419]: If not specify `aliases`, `name` will be useless.
How about add a validation to make sure `name` and `aliases` mast be specified together?

File: huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go [hunk 4]
```diff
@@ -514,3 +514,131 @@ resource "huaweicloud_fgs_function" "test" {
...
+  app                   = "default"
+  handler               = "index.handler"
+  memory_size           = 128
+  timeout               = 3
+  runtime               = "Python2.7"
+  code_type             = "inline"
+  func_code             = "dCA9ICdIZWxsbyBtZXNzYWdlOiAnICsganN="
+
+  versions {
+    name = "latest"
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_function_test.go:591]: Seems like nothing will happen for `versions` in this test function

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go (hunk 2 above)
Comment (deer-hang) [inline huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go:527]:
```suggestion
	for _, v := range aliasList {
		result[v.Version] = append(result[v.Version], map[string]interface{}{
			"name":        v.Name,
			"description": v.Description,
		})
	}
```

Comment (Lance52259) [reply huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go:527]: done.

**Pull request merged**